### PR TITLE
Add AxisMatrix alias

### PR DESCRIPTION
**Pull request by JeffFessler**
for consistency with `AbstractArray` and `AbstractMatrix` and other such types

### CODE REVIEW

File: src/core.jl [hunk 1]
```diff
@@ -178,6 +178,12 @@ struct AxisArray{T,N,D,Ax} <: AbstractArray{T,N}
     AxisArray{T,N,D,Ax}(data::AbstractArray{T,N}, axs::Tuple{Vararg{Axis,N}}) where {T,N,D,Ax} = new{T,N,D,Ax}(data, axs)
 end
 
+"""
+    AxisMatrix{T}
+Alias for [`AxisArray{T,2,D,Ax}`](@ref).
```
Comment (johnnychen94) [inline src/core.jl:183]:
```suggestion
Alias for [`AxisArray{T,2,D,Ax}`](@ref AxisArray).
```

**Review by johnnychen94 — COMMENTED** [review]
and maybe also `AbstractVector`?

**Review by johnnychen94 — APPROVED** [review]
I still have other high-priority tasks that need my care so I only merged the urgent one #201 made a patch release for it. Hope you understand.

Approval for the goal here but won't ship it very soon.

File: src/core.jl [hunk 2]
```diff
@@ -178,6 +178,12 @@ struct AxisArray{T,N,D,Ax} <: AbstractArray{T,N}
     AxisArray{T,N,D,Ax}(data::AbstractArray{T,N}, axs::Tuple{Vararg{Axis,N}}) where {T,N,D,Ax} = new{T,N,D,Ax}(data, axs)
 end
 
+"""
+    AxisMatrix{T}
+Alias for [`AxisArray{T,2,D,Ax}`](@ref AxisArray).
+"""
+const AxisMatrix{T,D,Ax} = AxisArray{T,2,D,Ax}
```
Comment (JeffFessler) [inline src/core.jl:185]:
```suggestion
const AxisMatrix{T,D,Ax} = AxisArray{T,2,D,Ax}
const AxisVector{T,D,Ax} = AxisArray{T,1,D,Ax}
```

**Comment by JeffFessler** [comment]
Thanks @johnnychen94 for handling the important one.  This one is not urgent at all...

**Comment by cmichelenstrofer** [comment]
Should `AxisMatrix` be exported?

**Comment by JeffFessler** [comment]
Yes, and `AxisVector` too I think.  That was an oversight on my part when I made the original PR.
Want to make a PR?

**Comment by cmichelenstrofer** [comment]
done #222 . Note that unlike `Vector` or `Matrix`, `AxisMatrix` and `AxisVector` don't work as constructors (A bunch of new constructors would have to be created mirroring those for `AxisArray`). They are still useful for limiting function argument types.

**Pull request merged**